ratpack/src/main/groovy/com/baeldung/Ratpack.groovy

```diff
@@ -0,0 +1,79 @@
+package com.baeldung;
+
+@Grab('io.ratpack:ratpack-groovy:1.6.1')
+import static ratpack.groovy.Groovy.ratpack
+
+import com.baeldung.model.User
+import com.google.common.reflect.TypeToken
+import ratpack.exec.Promise
+import ratpack.handling.Context
+import ratpack.jackson.Jackson
+import groovy.sql.Sql
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import ratpack.hikari.HikariModule
+import javax.sql.DataSource;
+
+ratpack {
+    serverConfig {
+        port(5050)
+    }
+    bindings {
+        module(HikariModule) { config ->
+            config.dataSourceClassName = 'org.h2.jdbcx.JdbcDataSource'
+            config.addDataSourceProperty('URL', "jdbc:h2:mem:devDB;INIT=RUNSCRIPT FROM 'classpath:/User.sql'")
+        }
+    }
+
+    handlers {
+
+        get { render 'Hello World from Ratpck with Groovy!!' }
+
+        get("greet/:name") { Context ctx ->
+            render "Hello " + ctx.getPathTokens().get("name") + "!!!"
+        }
+
+        get("data"){
+            render Jackson.json([title:"Mr",name:"Norman",country:"USA"])
+        }
+
+        post("user") {
+            Promise<User> user = parse(Jackson.fromJson(User))
+            user.then { u -> render u.name }
```

Comment:
Question - could this also be expressed without the local variable?

e.g.

```
parse(Jackson.fromJson(User))
    .then { user -> render user.name }
```

It might read more like a DSL if it can be made more fluent.

Reply:
Yes it is possible, however to keep the promise example easy I have not changed it.
For variation, I have used similar DSL structure in post('addUser') handler